### PR TITLE
tests: drivers: bbram: add test BBRAM on twister

### PR DESCRIPTION
**Pull request by Desvauxm-st**
add tests BBRAM on twister (on hw in CI)
add test for following boards:
nucleo_f429zi, nucleo_f207zg,nucleo_f303re,
nucleo_f746zg, nucleo_h743zi,nucleo_l152re,
disco_l475_iot1, nucleo_wb55rg

### CODE REVIEW

File: boards/arm/nucleo_h743zi/nucleo_h743zi.dts [hunk 1]
```diff
@@ -117,6 +117,9 @@ zephyr_udc0: &usbotg_fs {
 	clocks = <&rcc STM32_CLOCK_BUS_APB4 0x00010000>,
 		 <&rcc STM32_SRC_LSI RTC_SEL(2)>;
 	status = "okay";
+	backup_regs {
```
Comment (FRASTM) [inline boards/arm/nucleo_h743zi/nucleo_h743zi.dts:120]: add a line break

File: tests/drivers/bbram/testcase.yaml [hunk 2]
```diff
@@ -5,25 +5,45 @@ common:
...
       - it82xx2_evb
   drivers.bbram.npcx:
+    build_only: true
     filter: dt_compat_enabled("nuvoton,npcx-bbram")
     integration_platforms:
       - npcx9m6f_evb
   drivers.bbram.stm32:
     extra_args: OVERLAY_CONFIG="stm32.conf"
     filter: dt_compat_enabled("st,stm32-bbram")
+    platform_allow:
```
Comment (FRASTM) [inline tests/drivers/bbram/testcase.yaml:24]: `build_only: true` removed here is it correct ?

File: tests/drivers/bbram/testcase.yaml [hunk 3]
```diff
@@ -5,25 +5,45 @@ common:
...
+      - nucleo_h743zi
+      - nucleo_l152re
+      - disco_l475_iot1
+      - nucleo_wb55rg
     integration_platforms:
       - stm32f746g_disco
   drivers.bbram.stm32_rtc:
     extra_args: OVERLAY_CONFIG="stm32_rtc.conf"
     filter: dt_compat_enabled("st,stm32-bbram")
+    platform_allow:
```
Comment (FRASTM) [inline tests/drivers/bbram/testcase.yaml:39]: `build_only: true` removed here is it correct ?

File: tests/drivers/bbram/testcase.yaml [hunk 4]
```diff
@@ -5,25 +5,45 @@ common:
...
       - npcx9m6f_evb
   drivers.bbram.stm32:
     extra_args: OVERLAY_CONFIG="stm32.conf"
     filter: dt_compat_enabled("st,stm32-bbram")
+    platform_allow:
+      - stm32f746g_disco
+      - nucleo_f429zi
+      - nucleo_f207zg
+      - nucleo_f303re
+      - nucleo_f746zg
```
Comment (FRASTM) [inline tests/drivers/bbram/testcase.yaml:29]: is this one really needed ? --> already supporting stm32f746g_disco

File: tests/drivers/bbram/testcase.yaml [hunk 5]
```diff
@@ -5,25 +5,45 @@ common:
...
     integration_platforms:
       - stm32f746g_disco
   drivers.bbram.stm32_rtc:
     extra_args: OVERLAY_CONFIG="stm32_rtc.conf"
     filter: dt_compat_enabled("st,stm32-bbram")
+    platform_allow:
+      - stm32f746g_disco
+      - nucleo_f429zi
+      - nucleo_f303re
+      - nucleo_f746zg
```
Comment (FRASTM) [inline tests/drivers/bbram/testcase.yaml:43]: is this one really needed ? --> already supporting stm32f746g_disco

**Review by FRASTM — CHANGES_REQUESTED** [review]

**Comment by FRASTM** [comment]
This testcase is build only: Could we have a more pertinent testing like the sample : samples/boards/stm32/backup_sram/

**Comment by Desvauxm-st** [comment]
I will do another test into samples/boards/stm32/...

**Pull request closed without merging**